stdlib/2and3/ftplib.pyi

```diff
@@ -0,0 +1,123 @@
+from types import TracebackType
+from socket import socket
+from ssl import SSLContext
+
+MSG_OOB = ...  # type: int
+FTP_PORT = ...  # type: int
+MAXLINE = ...  # type: int
+CRLF = ...  # type: str
+if sys.version_info >= (3,):
+    B_CRLF = ...  # type: bytes
+
+class Error(Exception): ...
+class error_reply(Error): ...
+class error_temp(Error): ...
+class error_perm(Error): ...
+class error_proto(Error): ...
+
+T = TypeVar('T')
+
+class FTP(Generic[AnyStr]):
+    debugging = ...  # type: int
+    host = ...  # type: AnyStr
+    port = ...  # type: int
+    maxline = ...  # type: int
+    sock = ...  # type: Optional[socket]
+    welcome = ...  # type: Optional[str]
+    passiveserver = ...  # type: int
+    timeout = ...  # type: int
+    af = ...  # type: int
+    lastresp = ...  # type: str
+
+    if sys.version_info >= (3,):
+        file = ...  # type: Optional[TextIO]
+        encoding = ...  # type: str
+        def __enter__(self: T) -> T: ...
+        def __exit__(self, exc_type: Optional[type], exc_val: Optional[Exception],
+                     exc_tb: Optional[TracebackType]) -> bool: ...
+    else:
+        file = ...  # type: Optional[BinaryIO]
+
+    if sys.version_info >= (3, 3):
+        source_address = ...  # type: Optional[Tuple[str, int]]
+        def __init__(self, host: AnyStr = ..., user: str = ..., passwd: str = ..., acct: str = ...,
+                     timeout: float = ..., source_address: Optional[Tuple[str, int]] = ...) -> None: ...
+        def connect(self, host: AnyStr = ..., port: int = ..., timeout: float = ...,
+                    source_address: Optional[Tuple[str, int]] = ...) -> str: ...
+    else:
+        def __init__(self, host: AnyStr = ..., user: str = ..., passwd: str = ..., acct: str = ...,
+                     timeout: float = ...) -> None: ...
+        def connect(self, host: AnyStr = ..., port: int = ..., timeout: float = ...) -> str: ...
+
+    def getwelcome(self) -> str: ...
+    def set_debuglevel(self, level: int) -> None: ...
+    def debug(self, level: int) -> None: ...
+    def set_pasv(self, val: Union[bool, int]) -> None: ...
+    def sanitize(self, s: str) -> str: ...
+    def putline(self, line: str) -> None: ...
+    def putcmd(self, line: str) -> None: ...
+    def getline(self) -> str: ...
+    def getmultiline(self) -> str: ...
+    def getresp(self) -> str: ...
+    def voidresp(self) -> str: ...
+    def abort(self) -> str: ...
+    def sendcmd(self, cmd: str) -> str: ...
+    def voidcmd(self, cmd: str) -> str: ...
+    def sendport(self, host: str, port: int) -> str: ...
+    def sendeprt(self, host: str, port: int) -> str: ...
+    def makeport(self) -> socket: ...
+    def makepasv(self) -> Tuple[str, int]: ...
+    def ntransfercmd(self, cmd: str, rest: Optional[str] = ...) -> Tuple[socket, int]: ...
+    def transfercmd(self, cmd: str, rest: Optional[str] = ...) -> socket: ...
```

Comment:
Well the docstring says:

        Optional `rest' argument can be a string that is sent as the
        argument to a REST command.  This is essentially a server
        marker used to tell the server to skip over any data up to the
        given marker.

I could make it a `Union[str, int]` since that would work.